optional displaying the sensors location on the map

c7h opened the pull request:
## Description:

the sensor is now using the correct attributes (_latitued_ and _longitued_) which enables them to show up on the map. 

![example](https://lh4.googleusercontent.com/HCkyxMOZ0twO8FVi-ee7aGKA3-uqYGiR4JK8a_j5LKbWA6izLbfCHFVODY5YCzsemZJ69lcXAJQEmrdEIO6c=w1280-h944)


## ~Breaking change~

~The sensor attributes for GPS location used to be `lat` and `long`. This has been changed to `latitude` and `longitude` respectively.~

**Related issue (if applicable):** fixes #<home-assistant issue number goes here>

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** home-assistant/home-assistant.github.io#4680

## Example entry for `configuration.yaml` (if applicable):
```yaml

```

## Checklist:
  - [x] The code change is tested and works locally.

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [x] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [x] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [x] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [x] New files were added to `.coveragerc`.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

c7h committed:
Changed sensor attributes for GPS

the sensor is now using the correct attributes (latitued and
longitued) which enables them to show up on the map.

tinloaf reviewed (approved):
Looks good! This is, however, a breaking change. Could you write a 1-2 sentences describing this in a special "Breaking Change" paragraph in your original PR text?

fabaff reviewed (changes requested):
Showing a sensor on the map needs to be an optional feature because if the sensor isn't around the corner of your location the map will be loaded with a zoom level that everything fits.

We discussed this for the `iss` platform already deeply and agreed on using `show_on_map`.

tinloaf commented:
Makes sense. The "airvisual" sensor seems to use the same config option. Perhaps it would be beneficial to extract that into `const.py`.

c7h committed:
added option to display the sensor on the map

the configuration will be extended by the optional 'show_on_map' flag.
Default is not display the sensor on the map.

c7h commented:
@fabaff I've implemented the requested change. It makes sense to give the user an option to exclude the sensor from the map. The documentation will require an update now as well.

tinloaf commented:
Now it technically has less features than before if `CONF_SHOW_ON_MAP = False`, because the coordinates are not part of the attributes anymore. The `iss` platform solves this by setting the `lat` and `lon` attributes if `CONF_SHOW_ON_MAP = False`, and the `ATTR_LATITUDE` and `ATTR_LONGITUDE` if `CONF_SHOW_ON_MAP = True`.

If you were to implement it in the same way, and set the default for `CONF_SHOW_ON_MAP` to false, it even wouldn't be a breaking change anymore.

c7h commented:
That's a valid point. I will change it accordingly. Not breaking things is good 👍

c7h committed:
making the change non-breaking: old default behaviour

tinloaf commented:
Looks good now. :+1:

fabaff commented on homeassistant/components/sensor/luftdaten.py:
```diff
@@ -121,6 +126,9 @@ def device_state_attributes(self):
             }
+            if self._show_on_map:
+                attr[ATTR_LATITUDE] = self.luftdaten.data.meta['latitude']
+                attr[ATTR_LONGITUDE] = self.luftdaten.data.meta['longitude']
```
If `show_on_map` is `True` then we will have duplicate entries in the state card (lat/ and long/longitude). It should be either lat and long or latitude and longitude.

c7h replied:
OK - didn't know that that's a problem. I thought this way it will be always _non-breaking_.

fabaff replied:
It's not a problem but looks wired and could confuse people because the same data is shown twice. .`[ATTR_LATITUDE]` and `[ATTR_LONGITUDE]` are exposed by default. Don't append it to the existing attributes, just make two different dict in regard of the configuration.

c7h replied:
ok, agreed! 👍

c7h committed:
removed doubled attributes for lat and lon

c7h commented:
@fabaff ping

fabaff reviewed (approved):
Thanks :bird:
